configure.ac

```diff
@@ -135,6 +135,9 @@ AC_ARG_WITH([systemdsystemunitdir],
 AC_ARG_ENABLE(debug,
               [  --enable-debug      Enable debugging code.],, enable_debug="no")
 
+AC_ARG_ENABLE(extra-warnings,
+              [  --disable-extra-warnings      Disable extra compiler warnings.],, enable_extra_warnings="yes")
+
 AC_ARG_ENABLE(env-wrapper,
```

Comment:
Please don't remove -Wall from the default. Having to enable this option
for current and future developers is another thing we would need to look
after. It should work out of the box.

Either of these solutions work for me:
1) have two sets of warnings, one always enabled the other optionally
enabled.
2) have one set, but then enable them all.

Since I don't know the kind of warnings you want to enable option 1. should
be the safer bet. But number two works for me too. If a  warning option
does something I don't agree with we can always remove that later.

My priority is our of the box sane warnings without explicit configure
options.
On Apr 25, 2016 10:58, "Tamas Nagy" <notifications@github.com> wrote:

> In configure.ac
> <https://github.com/balabit/syslog-ng/pull/976#discussion_r60888809>:
>
> > @@ -135,6 +135,9 @@ AC_ARG_WITH([systemdsystemunitdir],
> >  AC_ARG_ENABLE(debug,
> >                [  --enable-debug      Enable debugging code.],, enable_debug="no")
> >
> > +AC_ARG_ENABLE(extra-warnings,
> > +              [  --disable-extra-warnings      Disable extra compiler warnings.],, enable_extra_warnings="yes")
> > +
> >  AC_ARG_ENABLE(env-wrapper,
>
> We agreed with @dnsjts <https://github.com/dnsjts> that we should default
> this to no, I will send a fixup soon. I want to add dozens of further
> warnings under this condition. Are you absolutely positive that we should
> separate -Wall from the rest? I would list all of them here (or
> optionally pass each of them from the configure command line, but that's a
> bit overkill for our purposes).
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/balabit/syslog-ng/pull/976/files/242a3970c44b320d2989752ef7d250d81e84ef81#r60888809>
>